components/Cases/CaseLink.tsx

```diff
@@ -29,8 +29,8 @@ const getLink = (
       return `/people/${caseFormData.mosaic_id}/records/${recordId}?is_historical=${caseFormData.is_historical}`;
     case 'Historical_Visit':
       return `/people/${caseFormData.mosaic_id}/visits/${recordId}?is_historical=${caseFormData.is_historical}`;
-    case 'Warning_Note':
-      return `/people/${caseFormData.mosaic_id}/warning-notes/${recordId}`;
+    case 'API_WarningNote':
+      return `/people/${caseFormData.mosaic_id}/warning-notes/${caseFormData.warning_note_id}`;
```

Comment:
If `warning_note_id` is typed as optional below (in `types.ts`) do we need to handle that case here, or is it more that our types don't accurately reflect the actual data we get (e.g. if `form_name_overall === "API_WarningNote"` then `warning_note_id` is always present)?

Reply:
I think it might be the latter, but my knowledge of typescript is still growing.
@anabebs @jhackett1

Reply:
I've just had a quick skim of the `types` file and it looks like we already have a pattern for doing this in a more structured way, which is great news! @Renny4Real did you want to pair over that? I can go over the details a bit more on a call